packages/flutter/lib/src/rendering/object.dart

```diff
@@ -3001,6 +3001,21 @@ mixin ContainerParentDataMixin<ChildType extends RenderObject> on ParentData {
 ///
 /// Provides a child model for a render object subclass that has a doubly-linked
 /// list of children.
+/// 
+/// The [ChildType] specifies the type of render object for the children,
+/// e.g. [RenderBox].
+/// For the parent data of the children, the [ParentDataType] is required to 
+/// extend [ContainerParentDataMixin], which provides the necessary functionality to
+/// grab the siblings of [firstChild] or [lastChild]. 
+/// This data will be accessible via [RenderObject.parentData] of the children if 
+/// the children are set up correctly in [RenderObject.setupParentData] of the
+/// parent render object (the one with this mixin).
+///
+/// If using [RenderBox] for the child type, you will usually want to make use of
+/// [RenderBoxContainerDefaultsMixin] and extend [ContainerBoxParentData] for the
+/// parent data.
+///
+/// Moreover, this is a default mixin for render objects returned to [MultiChildRenderObjectWidget].
```

Comment:
It is more than a default. It is required. `MultiChildRenderObjectElement` will actively cast its render object down to `ContainerRenderObjectMixin`. So you'll get runtime errors if you do not use this mixin.